crates/in_stream/src/ws.rs

```diff
@@ -281,6 +298,10 @@ impl<Sub: InStreamStd> InStream<&mut WsFrame, WsFrame> for InStreamWss<Sub> {
     }
 
     fn read(&mut self, data: &mut WsFrame) -> Result<usize> {
+        if self.is_in_failure() {
```

Comment:
@neonphog These are the bits I'm not 100% on. I guess this will definitely make everything fail for the duration but do we maybe want to do this another way?

Reply:
@willemolding - A couple other options:

- One would be to move this code to a lower level, and in Tcp and Mem streams, call [shutdown](https://doc.rust-lang.org/std/net/struct.TcpStream.html#method.shutdown) when your failure timing is triggered.
- Alternately, if we wanted to go for more encapsulation, you could create `InStreamFailure` and `InStreamListenerFailure` structs that would mostly be pass-through like Tls, but instead of encrypting / decrypting things it would randomly drop its sub-connection object, letting rust close the underlying streams.